### DISCUSSION TITLE
Problème avec une variable d'état externe

### DISCUSSION THREAD

**Discussion by WNGJAC** [thelfer/tfel#417] [Q&A]
Bonjour,

Je fait actuellement des calculs thermomécaniques sous Cast3M en faisant appel à une loi de comportement et des propriétés matériaux compilées avec MFront, cependant il s'avère que mon calcul est arreté dès le premier pas de temps car une des variables d'états externes prend une valeur totalement incohérente (hors de son domaine de validité) alors que cette même variable externe est correctement déclaré sur Cast3M...

Savez-vous d'où peut venir le problème ?

![image](https://github.com/thelfer/tfel/assets/94053562/7bd666e1-f7aa-4ed1-8885-a4bc9bde366a)

Bien cordialement,

Jacques

**Comment by thelfer** [comment]
Il me semble que nous avons corrigé un bug récemment là dessus avec l'équipe Cast3M, mais je ne me souviens pas des détails...
Est-il possible de partager le code Cast3M et la loi ?

**Reply by WNGJAC** [reply]
Il est probable que la version de Cast3M que j'utilise ne soit pas la bonne dans ce cas (version 2020).

Je suis désolé, mais je ne suis pas autorisé à partager ni le code Cast3M, ni la loi de comportement. En revanche, je peux donner quelques précisions sur l'erreur, la loi de comportement et le code Cast3M fonctionnent sans problème avec un maillage avec des éléments linéaires, c'est en voulant refaire le calcul avec des éléments quadratiques que l'erreur se produit. 

Merci pour votre réponse.

**Reply by thelfer** [reply]
@WNGJAC cela semble cohérent avec l'erreur dont j'ai parlé. Il y a, je crois, un boucle mal taillée dans l'opérateur VARI. Pouvez-vous faire le test avec la 2023 ?

**Reply by WNGJAC** [reply]
Bonjour, 

N'ayant pas Cast3M2023 d'installé et n'ayant pas les droits pour l'installer, je ne peux pas faire le test pour le moment.

**Reply by thelfer** [reply]
Merci. J'essaierai de faire le test à mon retour de congés.
